File: src/com/Torvald/Terrarum/ConsoleCommand/ToggleNoClip.java
package com.Torvald.Terrarum.ConsoleCommand;

import com.Torvald.Terrarum.Game;
import com.Torvald.Terrarum.Terrarum;

/**
 * Created by minjaesong on 16-01-19.
 */
public class ToggleNoClip implements ConsoleCommand {
    @Override
    public void execute(String[] args) {
        boolean status = Terrarum.game.getPlayer().isNoClip();

        Terrarum.game.getPlayer().setNoClip(!status);
        new Echo().execute("Set no-clip status to " + String.valueOf(!status));
    }

    @Override
    public void printUsage() {
        new Echo().execute("toggle no-clip status of player");
    }
}

File: src/net/torvald/terrarum/console/ToggleNoClip.kt
package net.torvald.terrarum.console

import net.torvald.terrarum.TerrarumGDX

/**
 * Created by minjaesong on 16-01-19.
 */
internal object ToggleNoClip : ConsoleCommand {
    override fun execute(args: Array<String>) {
        val status = TerrarumGDX.ingame!!.player!!.isNoClip()

        TerrarumGDX.ingame!!.player!!.setNoClip(!status)
        Echo("Set no-clip status to " + (!status).toString())
    }

    override fun printUsage() {
        Echo("toggle no-clip status of player")
    }
}
